x-pack/test/security_solution_api_integration/test_suites/detections_response/basic_essentials_license/rule_management/coverage_overview.ts

```diff
@@ -0,0 +1,585 @@
+
+  describe('@serverless @ess coverage_overview', () => {
+    beforeEach(async () => {
+      await deleteAllRules(supertest, log);
+    });
+
+    it('does NOT error when there are no security rules', async () => {
+      // Creates a non-security type rule
+      await alertingApi.createRule({
+        consumer: 'security',
+        name: 'Threshold rule',
+        ruleTypeId: 'observability.rules.custom_threshold',
+        params: {
+          criteria: [],
+          searchConfiguration: {
+            query: {
+              query: '',
+              language: 'kuery',
+            },
+            index: 'data-view-id',
+          },
+        },
+      });
+      const rule1 = await createRule(
+        supertest,
+        log,
+        getCustomQueryRuleParams({ threat: generateThreatArray(1) })
+      );
+
+      const body = await getCoverageOverview(supertest);
+
+      expect(body).to.eql({
+        coverage: {
+          T001: [rule1.id],
+          TA001: [rule1.id],
+          'T001.001': [rule1.id],
+        },
+        unmapped_rule_ids: [],
+        rules_data: {
+          [rule1.id]: {
+            activity: 'disabled',
+            name: 'Custom query rule',
+          },
+        },
+      });
+    });
+
+    describe('without filters', () => {
+      it('returns an empty response if there are no rules', async () => {
+        const body = await getCoverageOverview(supertest);
+
+        expect(body).to.eql({
+          coverage: {},
+          unmapped_rule_ids: [],
+          rules_data: {},
+        });
+      });
+
+      it('returns response with a single rule mapped to MITRE categories', async () => {
+        const rule1 = await createRule(
+          supertest,
+          log,
+          getCustomQueryRuleParams({ threat: generateThreatArray(1) })
+        );
+
+        const body = await getCoverageOverview(supertest);
+
+        expect(body).to.eql({
+          coverage: {
+            T001: [rule1.id],
+            TA001: [rule1.id],
+            'T001.001': [rule1.id],
+          },
+          unmapped_rule_ids: [],
+          rules_data: {
+            [rule1.id]: {
+              activity: 'disabled',
+              name: 'Custom query rule',
+            },
+          },
+        });
+      });
+
+      it('returns response with an unmapped rule', async () => {
+        const rule1 = await createRule(
+          supertest,
+          log,
+          getCustomQueryRuleParams({ threat: undefined })
+        );
+
+        const body = await getCoverageOverview(supertest);
+
+        expect(body).to.eql({
+          coverage: {},
+          unmapped_rule_ids: [rule1.id],
+          rules_data: {
+            [rule1.id]: {
+              activity: 'disabled',
+              name: 'Custom query rule',
+            },
+          },
+        });
+      });
+    });
+
+    describe('with filters', () => {
+      describe('search_term', () => {
+        it('returns response filtered by tactic', async () => {
```

Comment:
Just before I started reviewing this PR, I went through the test plan. I missed details in the test plan -- IMHO, half of the scenarios there are too generic. In contrast, these integration tests are very specific, well-named, and clearly tell how users can use the coverage overview feature. I wish the test plan was that specific too.